src/crypto/crypto_context.cc

```diff
@@ -221,15 +221,15 @@ int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
                                        issuer);
 }
 
-unsigned long LoadCertsFromFile(  // NOLINT(runtime/int)
+static unsigned long LoadCertsFromFile(  // NOLINT(runtime/int)
     std::vector<X509*>* certs,
     const char* file) {
   MarkPopErrorOnReturn mark_pop_error_on_return;
 
   auto bio = BIOPointer::NewFile(file, "r");
   if (!bio) return ERR_get_error();
 
-  while (X509* x509 = PEM_read_bio_X509(
+  while (X509* x509 = PEM_read_bio_X509_AUX(
```

Comment:
Actually we may need to do something different here for `--use-system-ca` - it seems `NODE_EXTRA_CA_CERTS` ignores auxiliary information. For system store though, it should probably still check EKU and KU, similar to the policy on other systems..

Reply:
On a second look, this does not seem to be necessary - the root CA does not need this check. It's only that on Windows and macOS there are additional user settings (usually through GUI) that needs to be honored (and the Windows API looks like a odd intersection on the certificate's EKU). On Linux at least there is no such thing, and according to RFC 5280 it's not necessary to check root CA like this.